x/btcstaking/keeper/msg_server.go

```diff
@@ -215,14 +215,14 @@ func (ms msgServer) CreateBTCDelegation(goCtx context.Context, req *types.MsgCre
 			btcutil.NewTx(parsedMsg.StakingTx.Transaction),
 			btccParams.BtcConfirmationDepth,
 			uint32(parsedMsg.StakingTime),
-			vp.Params.MinStakingTimeBlocks,
```

Comment:
could you double check the tests ? 

I have changed this back locally to `vp.Params.MinStakingTimeBlocks` and run the tests, and they did not catch this.

Reply:
Oh that's because the tests are for `VerifyInclusionProofAndGetHeight`. Seems we need to have a special case for testing `CreateBTCDelegation`

Reply:
In [d729190](https://github.com/babylonlabs-io/babylon/pull/289/commits/d7291907ad6ec64b5451d6b22138d0d03898cb86) I added a happy case to `TestDoNotAllowDelegationWithoutFinalityProvider` so that if you change `vp.Params.MinStakingTimeBlocks`, the test won't pass. But I don't think it's comprehensive enough. We can create an issue to add comprehensive test for `CreateBTCDelegation`

Reply:
I also noticed that we don't have a validation rule to ensure `minStakingTime > minUnbondingTime` in params. I think this rule is important. Otherwise, the [verification](https://github.com/babylonlabs-io/babylon/blob/85689cb2b644c4729a7aebd019583e0f8c03c29a/x/btcstaking/keeper/inclusion_proof.go#L65) will never pass if the user choose to stake with `minStakingTime`:

```
	// ensure staking tx's timelock has more than unbonding BTC blocks left
	if btcTip.Height+minUnbondingTime >= endHeight {
		return nil, types.ErrInvalidStakingTx.
			Wrapf("staking tx's timelock has no more than unbonding(=%d) blocks left", minUnbondingTime)
	}
```

Reply:
>But I don't think it's comprehensive enough. We can create an issue to add comprehensive test for CreateBTCDelegation

Sure, lets create issue for this 👍